cpp.mk

```diff
@@ -110,6 +110,7 @@ debug_$1 : $${BUILD_DIR}/bin/$1
 # Valgrind arguments for $1
 $1_VALGRIND_ARGS += --gen-suppressions=all --leak-check=full --track-origins=yes --malloc-fill=cc --free-fill=df
 
+.PHONY: $${BUILD_DIR}/$1/valgrind_output.xml
```

Comment:
It's a bit odd that the target is a file which really exists but which is marked as phony - could you not just remove this target and have the valgrind command run as the first step of make valgrind_check (which is already phony, and can now not have any dependencies)?